[EDA] When the last Contact of a Household is marked as Deceased, their name is removed from the Household Account

kdy618 opened the pull request:
# Critical Changes

# Changes
## Update to Automatic Household Naming
We've fixed an issue with the Automatic Household Naming functionality. Now, when you delete, specify "Deceased," or specify "Exclude from Household Name" for the last Contact in a Household Account and no other Contact exists, the Account name remains intact. For example, if John Doe is the last non-excluded Contact associated with a household and is then marked "Deceased," the Account will no longer incorrectly revert to "Household." Instead, it will remain "John Doe Household."

# Issues Closed
Closes #839

# New Metadata

# Deleted Metadata

# Testing Notes
**Pre-requisite:** 
1. In EDA Settings: 
- Ensure Account Model is set to Household. 
- Ensure that "Automatic Household Naming" is checked. 

**Scenario 1: If Household Account Name Format = {!LastName} Household and Contacts are marked deceased**  (You can choose whichever Account Name Format you wish to test with.) 

1. Find an existing Account with multiple contacts OR Create a new Contact. 

2. Once an Account is created, create a second or multiple Contact(s) on that existing Account with different last name(s). 

3. **Mark one of the Contact's record with "Deceased"**
- Result(s): 
a. Account Name is updated where the LastName of the Contact marked deceased is removed from the Account Name and is updated with other existing Contacts' LastName. 

ie. (1) If current Account name is Bond and Legend Household 
(2) Contact with LastName Bond was marked "Deceased"
(3) Result: Account Name is now updated to "Legend Household". 

b. Check the Contact marked deceased record and ensure that the following fields are checked.
- Do Not Contact
- Exclude from Household Formal Greeting
- Exclude from Household Informal Greeting
- Exclude from Household Name 
- Naming Exclusion (This field will have values for Informal, Formal, and Name)

4. **Mark the last of the Contact record with "Deceased"**
- Result(s): 
a. Account Name is not updated. The Account Name will contain the LastName of the Contact that was just marked deceased. 

ie. (1) If current Account name is Bond and Legend Household 
(2) Contact with LastName Bond was marked "Deceased"
(3) Contact with LastName "Legend" was marked "Deceased"
(3) Result: Account Name is not updated and should still be named "Legend Household". 

b.  Check the Contact marked deceased record and ensure that the following fields are checked.
- Do Not Contact
- Exclude from Household Formal Greeting
- Exclude from Household Informal Greeting
- Exclude from Household Name 
- Naming Exclusion (This field will have values for Informal, Formal, and Name)

**Scenario 2: If Household Account Name Format = {!LastName} Household and Contacts are marked Exclude from Household Naming**  (You can choose whichever Account Name Format you wish to test with.) 

1. Find an existing Account with multiple contacts OR Create a new Contact. 

2. Once an Account is created, create a second or multiple Contact(s) on that existing Account with different last name(s). 

3. **Mark one of the Contact's record with "Exclude From Household Naming"** 
- Result(s): 
a. Account Name is updated where the LastName of the Contact marked "Exclude From Household Naming" is removed from the Account Name. 

ie.  (1) If current Account name is Bond and Legend Household 
(2) Contact with LastName Bond was marked "Exclude from Household Naming"
(3) Result: Account Name is now updated to "Legend Household". 

4. Mark the last of the Contact record "Exclude from Household Naming" 
- Result(s): 
a. Account Name is not updated. The Account Name will contain the LastName of the Contact that was just marked deceased. 

ie. (1) If current Account name is Bond and Legend Household 
(2) Contact with LastName Bond was marked "Exclude from Household Naming"
(3) Contact with LastName "Legend" was marked "Exclude from Household Naming"
(3) Result: Account Name is not updated and should still be named "Legend Household". 

**Scenario 3: If Household Account Name Format = {!LastName} Household and Contacts are marked Exclude from Household Naming or Deceased and last Contact was deleted**  (You can choose whichever Account Name Format you wish to test with.) 

1. Find an existing Account with multiple contacts OR Create a new Contact. 

2. Once an Account is created, create a second or multiple Contact(s) on that existing Account with different last name(s). 

3. **Mark one of the Contact's record with "Exclude From Household Naming"** 
- Result(s): 
a. Account Name is updated where the LastName of the Contact marked "Exclude From Household Naming" is removed from the Account Name. 

ie.  (1) If current Account name is Bond, James and Legend Household 
(2) Contact with LastName Bond was marked "Exclude From Household Naming"
(3) Result: Account Name is now updated to "James and Legend Household". 

4. **Mark one of the Contact's record with "Deceased"** 
- Result(s): 
a. Account Name is updated where the LastName of the Contact marked "Deceased" is removed from the Account Name. 

ie.  (1) If current Account name is James and Legend Household 
(2) Contact with LastName James was marked "Deceased"
(3) Result: Account Name is now updated to "Legend Household". 

b.  Check the Contact marked deceased record and ensure that the following fields are checked.
- Do Not Contact
- Exclude from Household Formal Greeting
- Exclude from Household Informal Greeting
- Exclude from Household Name 
- Naming Exclusion (This field will have values for Informal, Formal, and Name)

4. Delete the last Contact associated with the Account. 
- Result(s): 
a. Account name is not updated. 

ie.  (1) If current Account name "Legend Household". 
(2) Contact with LastName Legend was deleted. 
(3) Result: Account Name is not updated and should still show "Legend Household". 

**Scenario 4: If Household Account Name Format = {!LastName} Household and Contacts are marked  Deceased and one of the Contact's Deceased was unchecked.**  (You can choose whichever Account Name Format you wish to test with.) 

1. Find an existing Account with multiple contacts OR Create a new Contact. 

2. Once an Account is created, create a second or multiple Contact(s) on that existing Account with different last name(s). 

3. **Mark one of the Contact's record with "Deceased"** 
- Result(s): 
a. Account Name is updated where the LastName of the Contact marked "Deceased" is removed from the Account Name. 

ie.  (1) If current Account name is James and Legend Household 
(2) Contact with LastName James was marked "Deceased"
(3) Result: Account Name is now updated to "Legend Household". 

b.  Check the Contact marked deceased record and ensure that the following fields are checked.
- Do Not Contact
- Exclude from Household Formal Greeting
- Exclude from Household Informal Greeting
- Exclude from Household Name 
- Naming Exclusion (This field will have values for Informal, Formal, and Name)

4. Uncheck the "Deceased" check on a Contact who has this checked. 
- Result(s): 
a. Account name is updated and the Contact who previously was marked deceased, their LastName is now in the Account name. 

ie.  (1) If current Account name "Legend Household". 
(2) Contact with LastName James was updated so that "Deceased" checkbox is now unchecked. 
(3) Result: Account Name is updated and is now "James and Legend Household". 

b.  Check the Contact marked deceased record and ensure that the following fields are unchecked.
- Do Not Contact
- Exclude from Household Formal Greeting
- Exclude from Household Informal Greeting
- Exclude from Household Name 
- Naming Exclusion (This field will have values for Informal, Formal, and Name)

**Scenario 5: If Household Account Name Format = {!LastName} Household and Contacts are marked Exclude from Household Naming  and one of the Contact's Exclude from Household Naming checkbox was unchecked.**  (You can choose whichever Account Name Format you wish to test with.) 

1. Find an existing Account with multiple contacts OR Create a new Contact. 

2. Once an Account is created, create a second or multiple Contact(s) on that existing Account with different last name(s). 

3. **Mark one of the Contact's record with "Exclude from Household Naming"** 
- Result(s): 
a. Account Name is updated where the LastName of the Contact marked "Exclude from Household Naming" is removed from the Account Name. 

ie.  (1) If current Account name is James and Legend Household 
(2) Contact with LastName James was marked "Exclude from Household Naming"
(3) Result: Account Name is now updated to "Legend Household". 

4. Uncheck the "Exclude from Household Naming" check on a Contact who has this checked. 
- Result(s): 
a. Account name is updated and the Contact who previously was marked Exclude from Household Naming, their LastName is now in the Account name. 

ie.  (1) If current Account name "Legend Household". 
(2) Contact with LastName James was updated so that "Exclude from Household Naming" checkbox is now unchecked. 
(3) Result: Account Name is updated and is now "James and Legend Household".

kdy618 commented:
@ryanguest , I added the condition that if Exclude_from_Household_Name__c == TRUE, remove it from the Account name, and if it's the last contact, it wouldn't remove it from the Account name. The question I have now is what if it's the last contact and the user deletes that contact and the rest of the contacts associated from the Account are either marked "deceased" or "excluded from household name"? What should the expected behavior/result be? Currently, I do not have that handled in the code; therefore, the Account is then just renamed to "Household".

kdy618 commented:
To Do: 

- [ ] Update method name checkDeceasedContact() to incorporate "Exclude From Household Name check". 
- [ ] Need to add test methods for "Exclude from Household Name check"

- [ ] 

- [ ]

lifewithryan commented:
> @lifewithryan , I added the condition that if Exclude_from_Household_Name__c == TRUE, remove it from the Account name, and if it's the last contact, it wouldn't remove it from the Account name. The question I have now is what if it's the last contact and the user deletes that contact and the rest of the contacts associated from the Account are either marked "deceased" or "excluded from household name"? What should the expected behavior/result be? Currently, I do not have that handled in the code; therefore, the Account is then just renamed to "Household".

I think if they delete the contact and the only remaining are excluded, we leave the name as is if at all possible. It will definitely look weird and if they run refresh hh names it will likely just wind up household, but there's only so much we can do for them.

kdy618 commented:
> > @lifewithryan , I added the condition that if Exclude_from_Household_Name__c == TRUE, remove it from the Account name, and if it's the last contact, it wouldn't remove it from the Account name. The question I have now is what if it's the last contact and the user deletes that contact and the rest of the contacts associated from the Account are either marked "deceased" or "excluded from household name"? What should the expected behavior/result be? Currently, I do not have that handled in the code; therefore, the Account is then just renamed to "Household".
> 
> I think if they delete the contact and the only remaining are excluded, we leave the name as is if at all possible. It will definitely look weird and if they run refresh hh names it will likely just wind up household, but there's only so much we can do for them.

Okie 👍

kdy618 commented:
> > > @lifewithryan , I added the condition that if Exclude_from_Household_Name__c == TRUE, remove it from the Account name, and if it's the last contact, it wouldn't remove it from the Account name. The question I have now is what if it's the last contact and the user deletes that contact and the rest of the contacts associated from the Account are either marked "deceased" or "excluded from household name"? What should the expected behavior/result be? Currently, I do not have that handled in the code; therefore, the Account is then just renamed to "Household".
> > 
> > 
> > I think if they delete the contact and the only remaining are excluded, we leave the name as is if at all possible. It will definitely look weird and if they run refresh hh names it will likely just wind up household, but there's only so much we can do for them.
> 
> Okie 👍

@lifewithryan , please feel free to retest this in your scratch org. I've added the logics to handle (1) if it's the last contact that's being deleted and all other contacts associated to the account are either marked deceased or excluded from HH naming, that it would not update the Account name.  (2) if it's not the last contact that's being deleted and there are other existing contacts that are not marked deceased or excluded from HH naming, that it would update the Account name appropriately.

kdy618 commented:
@lifewithryan , I'm not sure if this is the behavior we want but I noticed that if you check "Exclude from Household Naming" on a Contact  it does not update the msp field "Name exclusions" with the value "Name". However, if you check the "Deceased" checkbox on a Contact, it updates the msp field "Name exclusions" with all the values ie. Formal, Informal, Name. Is this a bug?

lifewithryan commented:
> @lifewithryan , I'm not sure if this is the behavior we want but I noticed that if you check "Exclude from Household Naming" on a Contact it does not update the msp field "Name exclusions" with the value "Name". However, if you check the "Deceased" checkbox on a Contact, it updates the msp field "Name exclusions" with all the values ie. Formal, Informal, Name. Is this a bug?

@kdy618 -- I don't think it is actually a bug as the Name exclusions is meant for handling things like the informal and formal greetings. (Which EDA really doesn't do much with). It's mainly used so that when sending correspondence to a household, we don't greet someone by mentioning a loved one who might have passed away, etc. I'm going to say we probably don't need to add it to that list when "Exclude from Household Naming" is checked.

kdy618 commented:
> > @lifewithryan , I'm not sure if this is the behavior we want but I noticed that if you check "Exclude from Household Naming" on a Contact it does not update the msp field "Name exclusions" with the value "Name". However, if you check the "Deceased" checkbox on a Contact, it updates the msp field "Name exclusions" with all the values ie. Formal, Informal, Name. Is this a bug?
> 
> @kdy618 -- I don't think it is actually a bug as the Name exclusions is meant for handling things like the informal and formal greetings. (Which EDA really doesn't do much with). It's mainly used so that when sending correspondence to a household, we don't greet someone by mentioning a loved one who might have passed away, etc. I'm going to say we probably don't need to add it to that list when "Exclude from Household Naming" is checked.

@lifewithryan gotcha, thank you for confirming. 👍

lifewithryan commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
 
     /*******************************************************************************************************
-    * @description the current user's Locale/Language
+    * @description The current user's Locale/Language
```
thank you for cleaning this up!!

lifewithryan commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+
+        if (primaryContactId.size() > 0) {
+            for (Account acc : [SELECT Primary_Contact__c,
+                                (SELECT Id, Name FROM Contacts WHERE Id IN :primaryContactId LIMIT 1)
```
same question here regarding the subquery (it could be that I'm just not there yet)

kdy618 replied:
@lifewithryan , from this line https://github.com/SalesforceFoundation/EDA/blob/0042f119431dcf0fe45ffe6417034d80332e4463/src/classes/ACCT_IndividualAccounts_TDTM.cls#L671 's query, it seems a bit redundant from line 659 and 665. However, this particular soql for loop is specifically querying for a Contact that's associated to the AccountId that matches the Primary_Contact__c value from the Account record. It then does a check if there's no Contact record that's associated to the AccountId that matches the Id on the Primary_Contact__c on the Account OR if the Primary_Contact__c field is blank on the Account.

lifewithryan commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+        }
+
+        List<Account> listAcc = [SELECT Id, RecordTypeId, Primary_Contact__c, OwnerId,
+                                (SELECT Id FROM Contacts limit 2)
```
I may need a walkthrough a bit with the subqueries. What if we have 3 contacts that come back, we aren't always guaranteed to get them in a particular order, etc. (Also "limit" to LIMIT). It may become more clear to me as I walk through more, so pardon the questions with these.

kdy618 replied:
@lifewithryan, I'm actually not sure why there's limit to these subqueries. This was the original subqueries so I'm not sure if there's a use case/reason that we're missing.

lifewithryan commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+
+        for (Account acc : listAcc) {
+            //TODO: use setting to determine if Account owner should be changed (W-009780)
+            if (acc.Contacts.size() == 1 && acc.Primary_Contact__c != NULL &&
```
Seems like we have a number of comparisons here as well. any chance we can make them more readable? for example, for me it is very easy for me to miss that last grouping. I like creating some booleans and passing those in instead since it allows you to be more explicit with the variable name, and a bit easier to read what conditions you're testing.

kdy618 replied:
@lifewithryan , sure will make the update for this.

lifewithryan commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+        }
+
+        accountsOnContacts = [SELECT Id, Type, RecordTypeId,
+                            (SELECT Id FROM Opportunities LIMIT 1),
```
Are we doing anything with Opportunties? (It's possible this is a GEM thing so perhaps we won't want to touch it, but I am going to tag Ben D or Luke P in an external comment here just in case for input. (cc: @lparrott or @bdvorachek -- are y'all using / needing this Opportunity subquery for GEM?)

bdvorachek replied:
@lifewithryan  I'm not aware of anything, but line 779 here is where that sub-select gets used.

lifewithryan replied:
aahhh thanks ben, I skimmed right past that apparently. :)

lifewithryan commented on src/classes/ACCT_IndividualAccounts_TEST.cls:
```diff
@@ -56,10 +55,12 @@ private class ACCT_IndividualAccounts_TEST {
         Test.stopTest();
 
-        Contact[] insertedContacts = [Select FirstName, LastName, AccountId  from Contact where id=:con.id];
+        Contact[] insertedContacts = [SELECT FirstName, LastName, AccountId
```
so nice to not have to scroll to the right! thank you!

lifewithryan commented on src/classes/ACCT_IndividualAccounts_TEST.cls:
```diff
@@ -133,7 +141,9 @@ private class ACCT_IndividualAccounts_TEST {
-        con = [select AccountId from Contact where id=:con.id];
+        con = [SELECT AccountId
+               FROM Contact
+               WHERE id = :con.id];
```
nit: id to Id :)

lifewithryan commented on src/classes/ACCT_IndividualAccounts_TEST.cls:
```diff
@@ -195,21 +210,25 @@ private class ACCT_IndividualAccounts_TEST {
-        Contact[] insertedContacts = [Select Account.Name, AccountId from Contact where id=:con.id];
+        Contact[] insertedContacts = [SELECT Account.Name, AccountId
+                                      FROM Contact
+                                      WHERE id = :con.Id];
```
nit: id to Id :)

lifewithryan commented on src/classes/ACCT_IndividualAccounts_TEST.cls:
```diff
@@ -534,20 +587,22 @@ private class ACCT_IndividualAccounts_TEST {
-        system.assertEquals('Household Account', assertAccount.RecordType.Name);
-        system.assertEquals(con.FirstName + ' ' + con.LastName + ' Household', assertAccount.Name);
+        Account assertAccount = [SELECT Id, RecordType.Name, Name
+                                 FROM Account limit 1];
```
nit: limit to LIMIT

lifewithryan commented on src/classes/ACCT_IndividualAccounts_TEST.cls:
```diff
@@ -605,30 +663,33 @@ private class ACCT_IndividualAccounts_TEST {
-        system.assertEquals('Household Account', assertAccount.RecordType.Name);
-        system.assertEquals(con.FirstName + ' ' + con.LastName + ' Household', assertAccount.Name);
+        Account assertAccount = [SELECT Id, RecordType.Name, Name
+                                 FROM Account limit 1];
```
nit: limit to LIMIT

lifewithryan commented on src/classes/ACCT_IndividualAccounts_TEST.cls:
```diff
@@ -685,30 +749,34 @@ private class ACCT_IndividualAccounts_TEST {
-        system.assertEquals(con.FirstName + ' ' + con.LastName + ' Household', assertAccount.Name);
+        assertAccount = [SELECT Id, RecordType.Name, Name
+                         FROM Account
+                         WHERE Id =: assertAccount.Id];
```
nit: just pointing out here that in some places we have "=: something" and in others we have "= :something" -- nothing I'd hold up a merge for.

kdy618 replied:
@lifewithryan , no prob! I'll clean it up.

lifewithryan commented on src/classes/UTIL_ACCT_Naming.cls:
```diff
@@ -35,296 +35,650 @@
 
         //Re-query to get correct Account values (and all the other fields we will look at)
-        listAccsToRename = database.query(strSoql);
+        returnedContacts = database.query(dynamicSoql);
```
consider using String.escapeSingleQuotes on the dynamicSoql query. The platform already does this for fields but it shows that we are indeed leveraging best practice.

kdy618 replied:
@lifewithryan  yep! Updated it to add that.

kdy618 commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
-                            mapContactIdAccountIdOldMoveOpps.put(c.Id, oldContacts[i].AccountId);
-                        }
-                    }
+        returnedContacts = database.query(dynamicSoql);
```
@lifewithryan , I can make the changes on this PR for that :)

kdy618 commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+        //Requery to retrieve correct Account value
+        relatedAccounts = [SELECT Id, RecordTypeID
+                           FROM Account
+                           WHERE Id IN : relatedAccountIds];
```
@lifewithryan , nope you are right. It's a miss. I'll update to make it consistent to follow :variable.

kdy618 commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+        //Retrieves all Accounts that have a connected Primary Contact
+        for (Account acc : [SELECT Id, Primary_Contact__c, RecordTypeId
+                            FROM Account
+                            WHERE Primary_Contact__c IN : contactIds])
```
updated the colon placement to be consistent to :variable

lifewithryan commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+
+        accountsOnContacts = [SELECT Id, Type, RecordTypeId,
+                            (SELECT Id FROM Opportunities LIMIT 1),
+                            (SELECT Id FROM Contacts LIMIT 1)
```
and these subqueries now make more sense to me :)

kdy618 commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+        * @param numOfContactsLeft is an interger that represents the number of non deceased Contacts associated
+        * with the Account.
+        */
+        private HandleUpdateWrapperLogic(Contact c, Contact oldContact, TDTM_Runnable.Action triggerAction, Integer numOfContactsLeft) {
```
@lifewithryan , for now I kept the multiple if statements in this method but cleaned up the if statements by creating new boolean variables for easy readability.

kdy618 commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+
+
+            //If Contact Name or Exclude_from_Household_Name__c value has changed
+            if ((c.LastName != oldContact.LastName ||
```
@lifewithryan , yep! Breaking out by using boolean variables for easy readability and reusability.

kdy618 commented:
@lifewithryan , recent updates for this branch contains updates from the bug fix for refactor account name branch.

kdy618 commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+
+        //Builds a set of primary Contact Ids from the Account given existing AccountIds
+        List<Account> listAccount = [SELECT Primary_Contact__c,
+                                     (SELECT Id, Name FROM Contacts LIMIT 1)
```
@lifewithryan , it looks like the subquery for this particular one is being utilized on this line https://github.com/SalesforceFoundation/EDA/blob/0042f119431dcf0fe45ffe6417034d80332e4463/src/classes/ACCT_IndividualAccounts_TDTM.cls#L699 where it's being used. As for the Contact Name, I don't see it being used. We can certainly removed the Name from being queried.

kdy618 commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,811 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+	* @return Id.
     */
-    private void insertContactAccount(Contact[] contacts, DmlWrapper dmlWrapper) {
+    private Id returnAccountRecordTypeId(Id accountId){
```
@lifewithryan , it looks like returnAccountRecordTypeId() method was created so that we can use it in the handlesAfterDelete() method on line 453. The only place I can see to retrieve the Account RecordTypeId is before handleAfterDelete() method is called on line 116.  This means I need to retrieve the Account RecordTypeId and create a separate method to query Contact's Account RecordTypeId for 'oldContacts' specifically since queryContacts() method which is called on line 94 is only for 'newContacts'.  Also, for handlesAfterDelete() method it is calling handleDeleteProcessing() method which is calling deleteContactAccountsIfEmpty() method and that's the only other one in this class that utilizes Account RecordTypeId but it's only comparing it is null or not. The other Account RecordTypeId comparison you're finding might be for handleInsertProcessing() and HandleUpdateWrapperLogic() which is pulling the Account recordtypeId from queryContacts() methods.

lifewithryan replied:
cool no changes needed :)

kdy618 commented on src/classes/ACCT_IndividualAccounts_TDTM.cls:
```diff
@@ -52,570 +48,811 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+        * with the Account.
+        */
+        private HandleUpdateWrapperLogic(Contact c, Contact oldContact, TDTM_Runnable.Action triggerAction, Integer numOfContactsLeft) {
+            Boolean conAccountIdNotNull = (c.AccountId != NULL) ? True : False;
```
@lifewithryan , remove the ternaries here. 👍

dgamboni commented:
Tested/Passes QE, will hold off merging till I get the go-ahead on RN's from @juliesnow (either here or via Quip)
